charts/consul/templates/_helpers.tpl

```diff
@@ -45,6 +49,38 @@ as well as the global.name setting.
             {{ "{{" }}- end -{{ "}}" }}
 {{- end -}}
 
+{{- define "consul.connectInjectWebhookTLSCertTemplate" -}}
+ |
+            {{ "{{" }}- with secret "{{ .Values.global.secretsBackend.vault.connectInject.tlsCert.secretName }}" "{{ printf "common_name=connect-injector.%s.%s" .Values.global.datacenter .Values.global.domain }}"
```

Comment:
we are not.  The common name currently in use is "Consul Webhook Certificates". I have tried to configure this including setting up the right role permissions in Vault, but still get `* common name Consul Webhook Certificates not allowed by this role` in `vault-agent-init` for both of those pods.  I'll have to revisit this.

Reply:
nvm. this is working.

Reply:
There is some logic in Vault that seems to tie `allowed_domains` with `common_name`.  I have set up a policy for `Consul Webhook Certificates Service` but still get this error.  It feels like there may be some logic that expects `common_name` to be something like the top level domain and interplays with `allowed_domains`.  A couple questions:
- Is `Consul Webhook Certificates Service` truly what is used as the common name with cert manager?
- If so, do we want to continue to use that across both controller and connect-injector?

Basically, what is our ideal common name pattern here?  It feels like I am running into a Vault configuration I need to set, so I want to make sure I trying to make this work with the ideal common name we are striving for because something that does not conform to a domain like pattern is going to be harder to troubleshoot.  So I just want to confirm and avoid unnecessary pain if possible.

Reply:
I think I might have mentioned it somewhere but could we make common name to be the name of the kubernetes service for the webhook. This will also be in the SANs and so should be allowed in the `allowed_domains`.

Reply:
Thank you.  Sorry I missed that.

I've made the name to match these two entries that are in the webhook-cert-manager configmap, since we changed the sans to match those in a previous comment.  
https://github.com/hashicorp/consul-k8s/blob/aa05d3db2158e0887f49a0f5354893603ff6b6f2/charts/consul/templates/webhook-cert-manager-configmap.yaml#L20
and
https://github.com/hashicorp/consul-k8s/blob/aa05d3db2158e0887f49a0f5354893603ff6b6f2/charts/consul/templates/webhook-cert-manager-configmap.yaml#L32

is this ok?

Reply:
Also, what is the reason that we make the sans and common name for server TLS include datacenter in it and not other certs like this?  Do we have concerns with mesh fed for server tls that does not apply to these other certs?